src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -425,7 +425,7 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
 
                 Vector256<float> scalarVector256 = Vector256.Create(scalar);
 
-                while (pDstCurrent + 8 <= pDstEnd)
+                while (pDstCurrent <= pDstEnd - 8)
                 {
                     Vector256<float> dstVector = Avx.LoadVector256(pDstCurrent);
```

Comment:
@tannergooding, do you know if there's any instruction with corresponds roughly to `addfloats xmm ptr [rax], xmm0` and if this trio of instructions collapses to that? It seems like it'd be more efficient and would avoid the xmm register spill.

Reply:
No, there is only load-forms, so this will compile down to two instructions (ideally):
```
addps tmp, scalarVector128, [pDstCurrent]
movps [pDstCurrent], tmp
```